packages/runtime-core/src/hydrationStrategies.ts

```diff
@@ -15,17 +15,17 @@ export type HydrationStrategy = (
   forEachElement: (cb: (el: Element) => any) => void,
 ) => (() => void) | void
 
-export type HydrationStrategyFactory<Options = any> = (
+export type HydrationStrategyFactory<Options> = (
   options?: Options,
 ) => HydrationStrategy
 
-export const hydrateOnIdle: HydrationStrategyFactory = () => hydrate => {
-  const id = requestIdleCallback(hydrate)
+export const hydrateOnIdle: HydrationStrategyFactory<number> = (timeout = 10000) => hydrate => {
```

Comment:
I think the default value of timeout is worth discussing. Maybe most people don't need to set the timeout parameter.

Reply:
Correct me if I'm wrong, but I believe that requestIdleCallback has an internal timeout if it isn't provided? I could theoretically make the logic a little more "mechanic" and conditionally provide the timeout, but it would be less clean. Would love to hear your thoughts about this!